src/training/vqa_fine_tune.py

```diff
@@ -0,0 +1,338 @@
+from tqdm import tqdm
+
+import open_clip
+from open_clip.factory import get_tokenizer
+from training.scheduler import cosine_lr
+from training.train import AverageMeter
+import evaluate
+from sklearn import preprocessing
+import numpy as np
+import sys
+
+from datasets import load_dataset_builder
+from datasets import load_dataset
+
+class VQATextDataset(Dataset):
+    def __init__(self, df, split, transforms, answer_set, tokenizer=None):
+        self.df = df
+        self.transforms = transforms
+        self.tokenize = tokenizer
+        self.num_classes = len(answer_set)
+    def __len__(self):
+        return len(self.df)
+
+    def __getitem__(self, idx):
+        item = self.df.iloc[idx]
+        img_path = item["image"]["path"]
+        image = Image.open(str(img_path))
+        text = item["question"]
+        target = np.zeros(self.num_classes)
+        for index, row in self.df.iterrows():
+            target[row['answer_list']] = row['answer_weights']
+        return {
+            'image': self.transforms(image),
+            'text': self.tokenize([text])[0],
+            'target': torch.tensor(target)
+        }
+
+def get_score(count: int) -> float:
+    return min(1.0, count / 3)
+
+def get_task_dataloaders(path, transforms, labelencoder, answer_set, args):
+    tokenizer = get_tokenizer(args.model)
+    dataloaders = {}
+    
+    for split in ["train", "validation"]:
+        dataset_train = load_dataset(path, split=split, cache_dir = "./sample_data")
+        dataset_df = dataset_train.to_pandas()
+
+        class_id = []
+        questions = []
+        images = []
+        answers = []
+        weights = []
+        for index, row in dataset_df.iterrows():
+            answer_count = {}
+            for answer in row['answers']:
+                answer_ = answer["answer"]
+                answer_count[answer_] = answer_count.get(answer_, 0) + 1
+            labels = []
+            scores = []
+            for answer in answer_count:
+                if answer not in answer_set:
+                    continue
+                labels.append(labelencoder.transform([answer])[0])
+                score = get_score(answer_count[answer])
+                scores.append(score)
+            if(len(labels) == 0):
+                continue
+            class_id.append(row['question_id'])
+            questions.append(row['question'])
+            images.append(row['image'])
+            answers.append(labels)
+            weights.append(scores)
+
+        class_id = np.array(class_id)
+        questions = np.array(questions)
+        images = np.array(images)
+        dataset_df = pd.DataFrame({'question_id': class_id, 'question': questions, 'image': images, 'answer_list': answers, 'answer_weights': weights})
+        #dataset_df = dataset_df[0:12800]
+        b_size = args.batch_size
+        if(split == "validation"):
+            b_size = args.batch_size * 20
+            dataset_df = dataset_df[0:12800]
```

Comment:
it's fine if you have this for debugging, but you should leave a `# TODO: ...` so we know to remove it before merging